lib/operations/add_user.js

```diff
@@ -1,186 +1,171 @@
 'use strict';
 
 const Aspect = require('./operation').Aspect;
-const defineAspects = require('./operation').defineAspects;
-const OperationBase = require('./operation').OperationBase;
 const CommandOperation = require('./command');
+const defineAspects = require('./operation').defineAspects;
 const applyWriteConcern = require('../utils').applyWriteConcern;
 const CONSTANTS = require('../constants');
 const count = require('./collection_ops').count;
 const crypto = require('crypto');
 const handleCallback = require('../utils').handleCallback;
 const loadDb = require('../dynamic_loaders').loadDb;
-const MongoError = require('../core').MongoError;
-const ReadPreference = require('../core').ReadPreference;
 const toError = require('../utils').toError;
 const updateOne = require('./collection_ops').updateOne;
+const WriteConcern = require('../write_concern');
 
-class AddUserOperation extends OperationBase {
+class AddUserOperation extends CommandOperation {
   constructor(db, username, password, options) {
-    super(options);
+    options = options || {};
+
+    // Special case where there is no password ($external users)
+    if (typeof username === 'string' && password != null && typeof password === 'object') {
```

Comment:
Should this positional logic be here? IMO that belongs in options management, so it should be outside the operation itself.